DEVELOPMENT.rst

```diff
@@ -111,14 +120,50 @@ reflect new commits.
 
 After your pull request is merged, you can fetch the new changes in the
 ``master`` branch or other branches you worked on, rebase your local clone of
-the repository and push them back to your own fork. Then, you can start working
-on another feature, or bugfix.
+the repository and push them back to your own fork, just as you would with any
+other commit from ``upstream``.
+Then, you can start working on another feature or bugfix.
+
+How to test your changes
+------------------------
+
+Once you push your contribution, Travis CI will run a first round of tests,
+mostly related to linting and syntax checking, then will promptly reject your
+contribution for the most pedantic reasons imaginable.
+
+In order to avoid this awkward scenario,
+you can (and should) run :command:`make test` yourself!
+
+But first, your development machine has to satisfy a few dependencies:
 
 .. code-block:: console
 
-   git checkout master
-   git fetch upstream
-   git rebase upstream/master
-   git push origin master
+   sudo apt install build-essential git graphviz shellcheck
+   pip install -U pycodestyle nose2 sphinx yamllint ansible-lint
+
+You also need **Docker**, and a user able to run docker containers.
```

Comment:
Oh, I had a bad installation of docker when I started, so I guess I confused docker's error message with an unmet dependency.

Personally I'd like it if docker wasn't involved at all in the first round of tests, since it looks like it hits the debian repos for a full upgrade every time it runs. Sounds more fitting for a proper testing environment with an apt mirror.

Also uh, having to uninstall it to skip it doesn't really make it optional.

I think `make docs` and `make docker` might be worth mentioning in the documentation. Would it make sense to change `make lint` to include all the linters and document that as well?

At any rate, I think I should mention `make help` in this guide.

Reply:
I suppose that the Docker test could be removed from `test-all` in the Makefile, and then called specifically on Travis-CI. I'll update that when I get back.